akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonRules.scala

```diff
@@ -465,7 +465,7 @@ private[parser] trait CommonRules { this: Parser with StringBuilding =>
     token ~> { s =>
       HttpMethods.getForKey(s) match {
         case Some(m) => m
-        case None    => HttpMethod.custom(s)
+        case None    => HttpMethod(s, isSafe = false, isIdempotent = false, requestEntityAcceptance = RequestEntityAcceptance.Expected, ConstantFun.anyToTrue)
```

Comment:
For custom headers received by the server, there is no need to validate them because they have already been parsed according to the spec, so skip the validation in `HttpMethod.custom`.